src/__experimental__/components/input/input.stories.js

```diff
@@ -0,0 +1,215 @@
+  groupedCharacterValue: '',
+  checkboxValue: false,
+  dateValue: '2009-12-12',
+  radioButtonGroupValue: ''
+});
+
+const getCommonProps = () => {
+  const percentageRange = {
+    range: true,
+    min: 0,
+    max: 100,
+    step: 1
+  };
+  const labelInline = boolean('labelInline', false);
+  const labelWidth = labelInline ? number('labelWidth', 30, percentageRange) : undefined;
+  const inputWidth = labelInline ? number('inputWidth', 70, percentageRange) : undefined;
+  const labelAlign = labelInline ? select('labelAlign', OptionsHelper.alignBinary) : undefined;
+
+  return {
+    labelInline,
+    labelWidth,
+    inputWidth,
+    labelAlign,
+    disabled: boolean('disabled', false),
+    readOnly: boolean('readOnly', false),
+    size: select('size', OptionsHelper.sizesRestricted, 'medium')
+  };
+};
+
+const InputIntegration = () => {
+  const commonProps = getCommonProps();
+  const disableFieldHelp = boolean('disable fieldHelp', false);
+  const disableLabelWidth = boolean('disable labelWidth (checkable inputs)', false);
+  const reverse = boolean('reverse (checkable inputs)', true);
+  const fieldHelpInline = disableFieldHelp ? '' : boolean('fieldHelpInline (checkable inputs)', false);
+
+  return (
+    <State store={ store }>
+      {state => <>
+        <Decimal
+          id='decimal'
+          label='Decimal Component'
+          fieldHelp={ disableFieldHelp ? '' : 'Decimal component fieldHelp' }
+          { ...commonProps }
+        />
+        <NumberInput
+          id='number'
+          label='Number Input Component'
+          onChange={ ev => store.set({ numberInputValue: ev.target.value }) }
+          fieldHelp={ disableFieldHelp ? '' : 'Number Input component fieldHelp' }
+          value={ state.numberInputValue }
+          { ...commonProps }
+        />
+        <GroupedCharacter
+          id='grouped-character'
+          label='Grouped Character Component'
+          separator='-'
+          groups={ [2, 2, 4] }
+          onChange={ ev => store.set({ groupedCharacterValue: ev.target.value.rawValue }) }
+          fieldHelp={ disableFieldHelp ? '' : 'Grouped Character Component fieldHelp' }
+          value={ state.groupedCharacterValue }
+          { ...commonProps }
+        />
+        <Select
+          ariaLabel='singleSelect'
+          label='Select Validation'
+          fieldHelp={ disableFieldHelp ? '' : 'Select Component fieldHelp' }
+          { ...commonProps }
+        >
+          <Option
+            key='option1'
+            text='Option 1'
+            value='option1'
+          />
+          <Option
+            key='option2'
+            text='Option 2'
+            value='option2'
+          />
+          <Option
+            key='option3'
+            text='Option 3'
+            value='option3'
+          />
+        </Select>
+        <Textarea
+          name='textarea'
+          label='Textarea Validation'
+          labelHelp='Returns an error when the field is empty'
```

Comment:
"Returns an error when the field is empty" – Is that relevant for this story?

Reply:
yup, it's a leftover from the original story.
changed.

Reply:
Thanks :white_check_mark: